### PR TITLE
Made haParams optional and added haSyncMode to PolicyDefinition

### PR DESCRIPTION
**Pull request by tylernap**

### CODE REVIEW

File: src/main/scala/com/dwolla/rabbitmq/policies/model.scala [hunk 1]
```diff
@@ -25,7 +25,6 @@ case class Policy(pattern: String,
                  )
 
 case class PolicyDefinition(haMode: String,
-                            haParams: Int,
```
Comment (staskolauren) [inline src/main/scala/com/dwolla/rabbitmq/policies/model.scala:28]: should we add `haSyncMode` to PolicyDefinition?

Comment (tylernap) [reply src/main/scala/com/dwolla/rabbitmq/policies/model.scala:28]: Yes! Totally forgot. I will add a commit here soon.

**Review by staskolauren — APPROVED** [review]
lgtm!

**Comment by tylernap** [comment]
Rethought this a little bit and decided to make haParams an optional field since, in order to make this generic, haParams is an optional field when haMode is something that is not "all". I also added haSyncMode to the policy definition model

**Comment by tylernap** [comment]
haParams was an Int and I made it a String. Fixed that back to Int as well

**Pull request merged**